checkov/terraform/parser.py

```diff
@@ -565,7 +568,26 @@ def parse_hcl_module(
 
         return module, tf_definitions
 
+    def _remove_unused_resolved_recursive(self, resolved):
+        for key in list(self.module_to_resolved.keys()):
+            file_key, module_index, module_name = key
+            if resolved != file_key:
+                continue
+            self._remove_unused_resolved(key)
+            break
+
+    def _remove_unused_resolved(self, key):
+        for new_resolved in self.module_to_resolved[key]:
```

Comment:
What if `key` does not exist in `self.module_to_resolved`?

Reply:
its shouldnt happen. if so, we need to investigate the case